### ISSUE TITLE
How can I use a proxy with ClientConnection

### ISSUE THREAD

**Issue by IlSassone** [grpc/grpc-swift#1389]
### What are you trying to achieve?

Describe your problem at a high level. Include code snippets if you think that
would help better illustrate your problem. Without enough information, we will
not be able to help you.

Is there a way to specify a proxy connection to use to connect to a grpc server?
This is the code that i'm currently using to instanciate the client used for the connection with the server
```swift
channel = ClientConnection.secure(group: self.group).connect(host: host, port: port)
client = GrpcClient(channel: channel!, defaultCallOptions: defaultCallOptions)
```
I would like to use a SOCS proxy to forward my calls to the grpc server, is there a way to do that?

**Comment by glbrntt** [comment]
Not out-of-the-box. However, [swift-nio-extras](https://github.com/apple/swift-nio-extras) includes a [SOCKS](https://github.com/apple/swift-nio-extras/tree/main/Sources/NIOSOCKS) library which you can use in conjunction with the `debugChannelInitializer`.

This would be something like:

```swift
import NIOSOCKS

// ...

channel = ClientConnection.secure(group: self.group)
  .withDebugChannelInitializer { channel in
    // Add the SOCKS client handler, it must be the first handler in the pipeline.
    channel.pipeline.addHandler(SOCKSClientHandler(targetAddress: ...), position: .first)
  }
  .connect(host: host, port: port)
```

**Comment by IlSassone** [comment]
Thanks for your reply, but it seems like i can't import directly `NIOSOCKS` like you mentioned
```swift
import NIOSOCKS // No such module 'NIOSOCKS'
import NIOExtras // No error, but i can't access 'SOCKSClientHandler'
```

**Comment by glbrntt** [comment]
You need to add a dependency to your Package.swift:

```swift
.package(url: "https://github.com/apple/swift-nio-extras.git", from: "1.9.0")
```

and then in the appropriate target where you're using `GRPC` you need to add:

```swift
    .target(
        name: "YourTargetWhichUsesGRPC",
        dependencies: [
            // ...
            .product(name: "NIOSOCKS", package: "swift-nio-extras"),
            // ...
        ]),

```

**Comment by IlSassone** [comment]
Oh ok, sorry but i'm no swift dev, the project that i'm working on is made with react native, and we are using swift for the [native-module](https://reactnative.dev/docs/native-modules-ios) part, so i guess we don't have a Package.swift file...

**Comment by glbrntt** [comment]
Yes, you can depend on gRPC Swift via Cocoapods _but_ -- and this is a big one -- gRPC Swift is about to stop supporting Cocoapods.

I believe you can still add a Swift Package Manager dependency to your workspace via File > Add Packages

**Comment by glbrntt** [comment]
For some reason there is no podspec published for NIOSOCKS.

We can publish one when we do the next release, however I would **strongly** advise you to **not** use Cocoapods as the next release will be the final release for which the NIO team will publish Cocoapods. The same is true of gRPC Swift: the next release will be the final one which supports Cocoapods.

**Comment by IlSassone** [comment]
I will really appreciate that, thanks for the support
